src/planner/builder/plan_node.go

```diff
@@ -64,3 +63,27 @@ func findLCA(h, p1, p2 SelectNode) SelectNode {
 	}
 	return pl
 }
+
+func findParent(tables []string, node SelectNode) SelectNode {
+	var parent SelectNode
+	for _, tb := range tables {
+		tbInfo := node.getReferTables()[tb]
+		if parent == nil {
+			parent = tbInfo.parent
+			continue
+		}
+		if parent != tbInfo.parent {
+			parent = findLCA(node, parent, tbInfo.parent)
+		}
+	}
+	return parent
+}
+
+func setFilter(s SelectNode, filter exprInfo) {
```

Comment:
setFilter-> addFilter is better?
because in the func, the operation is append, not Assignment

Reply:
good advice